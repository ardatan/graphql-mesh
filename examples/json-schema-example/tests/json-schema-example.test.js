const { findAndParseConfig } = require('@graphql-mesh/config');
const { getMesh } = require('@graphql-mesh/runtime');
const { readFile } = require('fs-extra');
const { basename, join } = require('path');

const { introspectionFromSchema, lexicographicSortSchema } = require('graphql');

const mesh$ = findAndParseConfig({
  dir: join(__dirname, '..'),
}).then(config => getMesh(config));

describe('JSON Schema Example', () => {
  it('should generate correct schema', async () => {
    const { schema } = await mesh$;
    expect(
      introspectionFromSchema(lexicographicSortSchema(schema), {
        descriptions: false,
      })
    ).toMatchSnapshot();
  });
  it('should give correct response', async () => {
    const { execute } = await mesh$;
    const query = await readFile(join(__dirname, '../example-query.graphql'), 'utf8');
    const result = await execute(query);
    expect(result?.data?.me?.firstName).toBeDefined();
    expect(result?.data?.me?.jobTitle).toBeDefined();
    expect(result?.data?.me?.lastName).toBeDefined();
    expect(result?.data?.me?.company?.name).toBeDefined();
    expect(result?.data?.me?.company?.type).toBeDefined();
    expect(result?.data?.me?.company?.employers).toHaveLength(2);
    expect(result?.data?.me?.company?.employers[0]?.firstName).toBeDefined();
    expect(result?.data?.me?.company?.employers[0]?.jobTitle).toBeDefined();
    expect(result?.data?.me?.company?.employers[0]?.lastName).toBeDefined();
  });
  afterAll(() => mesh$.then(mesh => mesh.destroy()));
});
