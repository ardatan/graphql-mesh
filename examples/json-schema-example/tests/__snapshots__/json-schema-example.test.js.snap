// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON Schema Example should generate correct schema 1`] = `
"scalar Any

\\"\\"\\"Fake Company\\"\\"\\"
type Company {
  employers: [User]
  name: String
  type: CompanyType
}

enum CompanyType {
  One_Person
  Private_Limited
  Public_Limited
}

input CreateUpdateUserInput {
  firstName: String
  jobTitle: String
  lastName: String
}

\\"\\"\\"Represents date values\\"\\"\\"
scalar Date

type Foo {
  id: String
}

type Mutation {
  createUser(input: CreateUpdateUserInput): User
  deleteUser(id: ID!): Any
  updateUser(id: ID!, input: CreateUpdateUserInput): User
}

type Query {
  company(id: ID!): Company
  me: User
}

\\"\\"\\"Fake User Object\\"\\"\\"
type User {
  birthDate: Date
  company: Company
  companyId: String
  firstName: String
  foos: [Foo]
  jobTitle: String
  lastName: String
}"
`;
